### PR TITLE
Add Docker rule to Makefile, update build documentation

### PR DESCRIPTION
**Pull request by bgammill**
*Description of changes:*
1. Added a Docker rule to the Makefile for users without Go locally installed.
2. Updated build documentation to reflect the options available for building.
3. Added explicit instructions for the build options.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,8 +8,17 @@ access to filesystems, and network connectivity.
 
 Building
 ---
+There are a few methods for building the binary.
+_We use [go modules](https://github.com/golang/go/wiki/Modules), so you need to build with Go 1.11 or newer._
 
-We use [go modules](https://github.com/golang/go/wiki/Modules), so you need to build with Go 1.11 or newer. `go build` or `make` should be sufficient to generate a working firectl binary.
+#### Make
+`make` is a Makefile rule which executes `go build`
```
Comment (samuelkarp) [inline README.md]:
```suggestion
The default Makefile rule executes `go build` and relies on the Go toolchain installed on your computer.
```

File: README.md [hunk 2]
```diff
@@ -8,8 +8,17 @@ access to filesystems, and network connectivity.
...
+
+`make docker` creates a temporary Docker container which builds and copies the binary to your current directory.
+#### Docker
```
Comment (samuelkarp) [inline README.md]: I think we can omit this section.  If someone wants to see how we use Docker, they can look at the definition of the `docker` target in the Makefile.

File: README.md [hunk 3]
```diff
@@ -8,8 +8,17 @@ access to filesystems, and network connectivity.
 
 Building
 ---
+There are a few methods for building the binary.
+_We use [go modules](https://github.com/golang/go/wiki/Modules), so you need to build with Go 1.11 or newer._
 
-We use [go modules](https://github.com/golang/go/wiki/Modules), so you need to build with Go 1.11 or newer. `go build` or `make` should be sufficient to generate a working firectl binary.
+#### Make
```
Comment (samuelkarp) [inline README.md]: I think we can remove the section heading, since I don't think we need a separate Make and Docker section.

**Comment by samuelkarp** [comment]
@bgammill Thanks for addressing some of my comments.  Are you planning to address the other comments as well?

**Comment by bgammill** [comment]
Should be good to go on those comments.

File: README.md (hunk 3 above)
Comment (samuelkarp) [reply README.md]: Can you remove the "#### Make" section heading?

File: README.md (hunk 2 above)
Comment (samuelkarp) [reply README.md]: Can you remove the entire "#### Docker" section?

**Comment by samuelkarp** [comment]
@bgammill It looks like you have not addressed my remaining comments.  Can you do so?

**Comment by samuelkarp** [comment]
@bgammill Thank you so much for your contribution! I've carried over your changes into https://github.com/firecracker-microvm/firectl/pull/38.

**Pull request closed without merging**